engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/batch/NadelBatchHydrationByIndex.kt

```diff
@@ -0,0 +1,114 @@
+package graphql.nadel.enginekt.transform.hydration.batch
+
+import graphql.nadel.ServiceExecutionResult
+import graphql.nadel.enginekt.blueprint.NadelBatchHydrationFieldInstruction
+import graphql.nadel.enginekt.transform.hydration.NadelHydrationUtil
+import graphql.nadel.enginekt.transform.hydration.NadelHydrationUtil.getNumberOfInputNodes
+import graphql.nadel.enginekt.transform.hydration.batch.NadelBatchHydrationInputBuilder.getBatchInputDef
+import graphql.nadel.enginekt.transform.result.NadelResultInstruction
+import graphql.nadel.enginekt.transform.result.json.JsonNode
+import graphql.nadel.enginekt.util.AnyList
+import graphql.nadel.enginekt.util.isList
+import graphql.nadel.enginekt.util.listOfNulls
+import graphql.nadel.enginekt.util.subListOrNull
+import graphql.nadel.enginekt.util.unwrapNonNull
+
+internal object NadelBatchHydrationByIndex {
```

Comment:
This is the main change, also moved this code out of the `NadelBatchHydrator` class as there were two batch hydration algorithms in there that were polluting the class.